src/topology/algebra/order/lower_topology.lean

```diff
@@ -0,0 +1,149 @@
+
+namespace lower_topology
+
+variable {α}
+
+lemma is_open_iff_generate_Ici_comp {s : set (lower α)} :
+  is_open s ↔ generate_open {s | ∃ a, (Ici a)ᶜ = s} s := iff.rfl
+
+/--
+Left-closed right-infinite intervals [a,∞) are closed in the lower topology
+-/
+lemma is_closed_Ici (a : lower α) : is_closed (Ici a) :=
+is_open_compl_iff.1 $ generate_open.basic _ ⟨a, rfl⟩
+
+/--
+The upper closure of a finite subset is closed in the lower topology
+-/
+lemma is_closed_upper_closure (F : set (lower α)) (h : F.finite) :
+  is_closed (upper_closure F : set (lower α)) :=
+begin
+  simp only [← upper_set.infi_Ici, upper_set.coe_infi],
+  exact is_closed_bUnion h (λ a h₁, is_closed_Ici a),
+end
+
+/--
+Every subset open in the lower topology is a lower set
+-/
+lemma is_lower_set_of_is_open {s : set (lower α)} (h : is_open s) : is_lower_set s :=
+begin
+  rw is_open_iff_generate_Ici_comp at h,
+  induction h,
+  case generate_open.basic : u h { obtain ⟨a, rfl⟩ := h, exact (is_upper_set_Ici a).compl },
+  case univ : { exact is_lower_set_univ },
+  case inter : u v hu1 hv1 hu2 hv2 { exact hu2.inter hv2 },
+  case sUnion : _ _ ih { exact is_lower_set_sUnion ih },
+end
+
+lemma is_upper_of_is_closed {s : set (lower α)} (h : is_closed s) : is_upper_set s :=
+is_lower_set_compl.1 $ is_lower_set_of_is_open h.is_open_compl
+
+/--
+The closure of a singleton {a} in the lower topology is the left-closed right-infinite interval
+[a,∞)
+-/
+lemma closure_singleton (a : lower α) : closure {a} = Ici a :=
+subset_antisymm (closure_minimal (λ b h, h.ge) $ is_closed_Ici a) $
+  (is_upper_of_is_closed is_closed_closure).Ici_subset (subset_closure rfl)
+
+/--
+The non-empty complements of the upper closures of finite subsets are a collection of lower sets
+which form a basis for the lower topology
+-/
+def lower_basis (α : Type u) [preorder α] :=
+{s : set α | ∃ (F : set α), F.finite ∧ ↑(upper_closure F).compl = s}
+
+lemma is_topological_basis : is_topological_basis (lower_basis (lower α)) :=
+begin
+  convert is_topological_basis_of_subbasis rfl,
+  simp_rw [lower_basis, upper_set.coe_compl, coe_upper_closure, compl_set_of],
+  push_neg, simp_rw set_of_forall,
+  ext s, split,
+  { rintro ⟨F, hF, rfl⟩,
+    refine ⟨(λ a, (Ici a)ᶜ) '' F, ⟨hF.image _, image_subset_iff.2 $ λ _ _, ⟨_, rfl⟩⟩, _⟩,
+    rw sInter_image, refl },
+  { rintro ⟨F, ⟨hF, hs⟩, rfl⟩,
+    haveI := hF.to_subtype,
+    rw [subset_def, subtype.forall'] at hs,
+    choose f he using hs,
+    refine ⟨_, finite_range f, set.ext $ λ a, _⟩,
+    simp_rw [bInter_range, Inter_subtype, ←compl_set_of, Ici_def, he, mem_Inter₂],
+    refl },
+end
+
+end lower_topology
+
+end pre_order
+
+section partial_order
+
+variables (α) [partial_order α]
+
+instance : partial_order (lower α) := ‹partial_order α›
+
+/-
+The lower topology on a partial order is T₀.
+-/
+@[priority 90] -- see Note [lower instance priority]
+instance lower_topology.to_t0_space : t0_space (lower α) :=
```

Comment:
This should be called `lower.t0_space`.
```suggestion
instance : t0_space (lower α) :=
```

Reply:
Note that it's no longer in a namespace; it would get the name `lower.t0_space` right (or whatever we change `lower` to)? Sounds good.

Reply:
I think the name gets generated automatically (`#check with_lower_topology.t0_space` afterwards is okay)